File: day1/csharp/calories.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Calories
{
    class Elf
    {
        public int calories = 0;
        public int position = 0;
    }
    class calories
    {

        static void Main()
        {
            string[] lines = System.IO.File.ReadAllLines(@"../../../input.txt");
            List<Elf> elves = new List<Elf>();
            int elfcount = 1;
            Elf currentElf = new Elf();
            currentElf.position = elfcount;
            foreach (string line in lines)
            {
                if (line != "")
                {
                    currentElf.calories += int.Parse(line);
                }
                else
                {
                    elves.Add(currentElf);
                    elfcount++;
                    currentElf = new Elf();
                    currentElf.position = elfcount;
                }
            }
            elves.Sort((a, b) => a.calories.CompareTo(b.calories));
            elves.Reverse();
            Console.WriteLine("Highest Count:" + elves.First().calories);
            Console.WriteLine("Top 3:" + elves.Take(3).Sum(a => a.calories));
            Console.ReadLine();
        }
    }
}

File: day1/python/calories.py
with open("../input.txt", "r") as input:
    input_lines = input.readlines()

elves = []
elf_count = 0
for line in input_lines:
    if line.strip():
        try:
            elves[elf_count] += int(line.strip())
        except IndexError:
            elves.append(int(line.strip()))
    else:
        elf_count += 1

sorted_elves = sorted(elves, reverse=True)
# Part 1 answer
print(f"Max calories: {sorted_elves[0]}")
# Part 2 answer
print(f"Top 3 sum: {sum(sorted_elves[0:3])}")
